### PR TITLE
bpo-31455: fix reference leak

### PR DESCRIPTION
**Pull request by scoder**
Fixes a reference leak to 'self' in the previous implementation.

https://bugs.python.org/issue31455

### CODE REVIEW

**Comment by scoder** [comment]
The travis build failure seems unrelated. The relevant tests had already passed by then.

File: Modules/_elementtree.c [hunk 1]
```diff
@@ -2575,14 +2575,24 @@ xmlparser(PyObject* self_, PyObject* args, PyObject* kw)
         return NULL;
     }
 
+    ALLOC(sizeof(XMLParserObject), "create expatparser");
+
+    /* Init to NULL to keep the error handling below manageable. */
+    self->target =
```
Comment (serhiy-storchaka) [inline Modules/_elementtree.c:2581]: This is not needed. `PyObject_New()` clears the memory.

Comment (scoder) [reply Modules/_elementtree.c:2581]: I thought so, too, until the test started crashing due to uninitalised struct fields.

Comment (serhiy-storchaka) [reply Modules/_elementtree.c:2581]: Oh, seems I remember this wrong.

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by serhiy-storchaka** [comment]
Seems the test is raised for unrelated cause.

**Pull request merged**